jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1318,11 +1318,19 @@ public void succeeded()
             {
                 callback = _writeCallback;
                 _writeCallback = null;
-                httpChannel = _request.lockedGetHttpChannelState();
-                httpChannel.lockedStreamSendCompleted(true);
+                httpChannel = _request._httpChannelState;
+                if (httpChannel != null)
+                    httpChannel.lockedStreamSendCompleted(true);
             }
+
             if (callback != null)
-                httpChannel._writeInvoker.run(callback::succeeded);
+            {
+                if (httpChannel != null)
+                    httpChannel._writeInvoker.run(callback::succeeded);
+                else
+                    // Channel already completed, just fail the callback.
+                    HttpChannelState.failed(callback, new IllegalStateException("channel already completed"));
```

Comment:
I'd use `IOException` here, as it is a valid code path.

Reply:
Done.